bridge/ffi/src/lib.rs

```diff
@@ -9,12 +9,42 @@
 //! To use the FFI layer, you need to build the dynamic library (e.g., .so, .dylib, or .dll) by adding
 //! the appropriate configuration to `Cargo.toml`. Then, include the generated dynamic library and
 //! the C header file `dsnp_graph_sdk_ffi.h` in your C, C++, or other FFI-compatible projects.
+
 mod c_api;
 pub use c_api::*;
 mod bindings;
 pub use bindings::*;
 mod utils;
 pub use utils::*;
+mod errors;
+pub use errors::*;
 
 #[cfg(test)]
 mod tests;
+
+use std::ptr::NonNull;
+
+#[repr(C)]
+#[derive(Debug, Clone, PartialEq)]
+pub struct FFIResult<T, E> {
```

Comment:
It seems like only one of result, error will ever be Some(value), while the other would be None. If that's the case, would it make sense for this to be a union instead of a struct?

Reply:
i like the suggestion but then we cant derive fancy things on unions like `Debug` etc? especially debug trait is ideal here

Open to changing it if preferred

Reply:
Just realized it will be warranting me to change a lot around so we should do it as an enhancement